build-gnulib-gdbserver/import/string.h:683:1: error: ambiguating new declaration of 'char* strchrnul(const char*, int)'

### Issue submission guidelines
See [detaled guidelines](http://crosstool-ng.github.io/support/). In short:
- Report crosstool-NG version
**crosstool-NG 1.23.0 (GNU Make 4.1 Built for i686-pc-linux-gnu ...)**
***
- Report host OS and its version
**Linux zzuli.ailab 4.16.1-041601-generic #201804081334 SMP Sun Apr 8 13:44:23 UTC 2018 i686 i686 i686 GNU/Linux**
***
- Attach `.config` (unless the issue is not related to any configuration)
**just type:` ./ct-ng aarch64-unknown-linux-android`**
***
- Attach `build.log` (unless crosstool-NG fails before the build starts)
At the beginning,all is well.Unfortunately,
When the build was in step 'Installing gdbserver',I got this error:
```
...
[ALL  ]    make[2]: Leaving directory '/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/build/build-gdb-gdbserver'
[ALL  ]    aarch64-unknown-linux-android-g++ -x c++      -I. -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../regformats -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/.. -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../../include -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../gnulib/import -Ibuild-gnulib-gdbserver/import  -Wall -Wpointer-arith -Wno-unused -Wunused-value -Wunused-function -Wno-switch -Wno-char-subscripts -Wempty-body -Wunused-but-set-parameter -Wunused-but-set-variable -Wno-sign-compare -Wno-narrowing -Wno-error=maybe-uninitialized  -DGDBSERVER -c -o agent.o -MT agent.o -MMD -MP -MF ./.deps/agent.Tpo /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common/agent.c
[ALL  ]    aarch64-unknown-linux-android-g++ -x c++      -I. -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../regformats -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/.. -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../../include -I/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../gnulib/import -Ibuild-gnulib-gdbserver/import  -Wall -Wpointer-arith -Wno-unused -Wunused-value -Wunused-function -Wno-switch -Wno-char-subscripts -Wempty-body -Wunused-but-set-parameter -Wunused-but-set-variable -Wno-sign-compare -Wno-narrowing -Wno-error=maybe-uninitialized  -DGDBSERVER -c -o ax.o -MT ax.o -MMD -MP -MF ./.deps/ax.Tpo `echo " -Wall -Wpointer-arith -Wno-unused -Wunused-value -Wunused-function -Wno-switch -Wno-char-subscripts -Wempty-body -Wunused-but-set-parameter -Wunused-but-set-variable -Wno-sign-compare -Wno-narrowing -Wno-error=maybe-uninitialized " | sed "s/ -Wformat-nonliteral / -Wno-format-nonliteral /g"` /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/ax.c
[ALL  ]    In file included from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common/common-defs.h:52,
[ALL  ]                     from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common/agent.c:20:
[ERROR]    build-gnulib-gdbserver/import/string.h:683:1: error: ambiguating new declaration of 'char* strchrnul(const char*, int)'
[ALL  ]     _GL_FUNCDECL_SYS (strchrnul, char *, (char const *__s, int __c_in)
[ALL  ]     ^~~~~~~~~~~~~~~~
[ALL  ]    In file included from build-gnulib-gdbserver/import/string.h:41,
[ALL  ]                     from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common/common-defs.h:56,
[ALL  ]                     from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common/agent.c:20:
[ALL  ]    /home/test0/x-tools/aarch64-unknown-linux-android/aarch64-unknown-linux-android/sysroot/usr/include/string.h:77:26: note: old declaration 'const char* strchrnul(const char*, int)'
[ALL  ]     extern "C++" const char* strchrnul(const char* __s, int __ch) __RENAME(strchrnul) __attribute_pure__ __INTRODUCED_IN(24);
[ALL  ]                              ^~~~~~~~~
[ALL  ]    In file included from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common/common-defs.h:52,
[ALL  ]                     from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/server.h:22,
[ALL  ]                     from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/ax.c:19:
[ERROR]    build-gnulib-gdbserver/import/string.h:683:1: error: ambiguating new declaration of 'char* strchrnul(const char*, int)'
[ALL  ]     _GL_FUNCDECL_SYS (strchrnul, char *, (char const *__s, int __c_in)
[ALL  ]     ^~~~~~~~~~~~~~~~
[ALL  ]    In file included from build-gnulib-gdbserver/import/string.h:41,
[ALL  ]                     from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/../common/common-defs.h:56,
[ALL  ]                     from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/server.h:22,
[ALL  ]                     from /home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/src/gdb/gdb/gdbserver/ax.c:19:
[ALL  ]    /home/test0/x-tools/aarch64-unknown-linux-android/aarch64-unknown-linux-android/sysroot/usr/include/string.h:77:26: note: old declaration 'const char* strchrnul(const char*, int)'
[ALL  ]     extern "C++" const char* strchrnul(const char* __s, int __ch) __RENAME(strchrnul) __attribute_pure__ __INTRODUCED_IN(24);
[ALL  ]                              ^~~~~~~~~
[ALL  ]    Makefile:584: recipe for target 'agent.o' failed
[ERROR]    make[1]: *** [agent.o] Error 1
[ERROR]    make[1]: *** Waiting for unfinished jobs....
[ALL  ]    Makefile:537: recipe for target 'ax.o' failed
[ERROR]    make[1]: *** [ax.o] Error 1
[ALL  ]    make[1]: Leaving directory '/home/test0/github/crosstool-ng/.build/aarch64-unknown-linux-android/build/build-gdb-gdbserver'
...    
```
I think that there are some problems about lib-version.However,I donn't know how to handle it.If you have any ideas,please tell me and thanks for you!

--- a/scripts/build/libc/bionic.sh
+++ b/scripts/build/libc/bionic.sh
@@ -26,7 +26,13 @@ do_libc() {
     fi
     CT_DoStep INFO "Installing C library binaries"
     CT_DoExecLog ALL cp -r "${CT_SRC_DIR}/android-ndk/platforms/android-${CT_ANDROID_API}/arch-${arch}/usr" "${CT_SYSROOT_DIR}"
-    CT_EnvModify CT_ALL_TARGET_CFLAGS "${CT_ALL_TARGET_CFLAGS} -D__ANDROID_API__=${CT_ANDROID_API}"
+
+    # NB: Modifying CT_TARGET_CFLAGS here, not CT_ALL_TARGET_CFLAGS: the __ANDROID_API__
+    # definition needs to be passed into GCC build, or the resulting libstdc++ gets
+    # miscompiled (attempt to link against it results in unresolved symbols to stdout/...).
+    # And since __ANDROID_API__ is a user config option, placing it with other user-supplied
+    # options isn't completely out of character.
+    CT_EnvModify CT_TARGET_CFLAGS "${CT_TARGET_CFLAGS} -D__ANDROID_API__=${CT_ANDROID_API}"
 }
 
 do_libc_post_cc() {

